tracer/build/_build/Build.Utilities.cs

```diff
@@ -213,4 +224,96 @@ partial class Build
+
+            var artifacts = await buildHttpClient.GetArtifactsAsync(
+                             project: AzureDevopsProjectId,
+                             buildId: buildNumber);
+
+            foreach(var artifact in artifacts)
+            {
+                if (!artifact.Name.Contains("snapshots"))
+                {
+                    continue;
+                }
+
+                var tempZipLocation = Path.GetTempFileName();
+                var extractLocation = Path.Combine(Path.GetDirectoryName(tempZipLocation), Path.GetFileNameWithoutExtension(tempZipLocation));
+
+                Logger.Info($"Unzipping '{artifact.Resource.DownloadUrl}' in '{extractLocation}'");
+
+                using var response = await client.GetAsync(artifact.Resource.DownloadUrl);
+                await using var fs = new FileStream(tempZipLocation, FileMode.Create);
+                await response.Content.CopyToAsync(fs);
+
+                ZipFile.ExtractToDirectory(tempZipLocation, extractLocation);
+
+                var snapshotsDirectory = Path.Combine(TestsDirectory, "snapshots");
+                var di = new DirectoryInfo(extractLocation);
+
+                Logger.Info($"Moving received snapshots from this archive to'{snapshotsDirectory}'");
+                foreach (var file in di.GetFiles("*", SearchOption.AllDirectories))
+                {
+                    var fileName = Path.GetFileNameWithoutExtension(file.FullName);
+                    if (fileName.EndsWith("received"))
+                    {
+                        var destFileName = Path.Combine(snapshotsDirectory, file.Name);
+                        if (!File.Exists(destFileName))
+                        {
+                            file.CopyTo(destFileName, overwrite: false);
+                        }
+                    }
+                }
+
+                File.Delete(tempZipLocation);
+                Directory.Delete(extractLocation, recursive: true);
+            }
+
+            ReplaceReceivedFilesInSnapshots();
+      });
+
+    private void ReplaceReceivedFilesInSnapshots()
+    {
+        var snapshotsDirectory = Path.Combine(TestsDirectory, "snapshots");
+        var directory = new DirectoryInfo(snapshotsDirectory);
+        var files = directory.GetFiles("*.received*");
+
+        var suffixLength = "received".Length;
+        foreach (var file in files)
+        {
+            var source = file.FullName;
+            var fileName = Path.GetFileNameWithoutExtension(source);
+            if (!fileName.EndsWith("received"))
+            {
+                Logger.Warn($"Skipping file '{source}' as filename did not end with 'received'");
+                continue;
+            }
+            var trimmedName = fileName.Substring(0, fileName.Length - suffixLength);
+            var dest = Path.Combine(directory.FullName, $"{trimmedName}verified{Path.GetExtension(source)}");
+            file.MoveTo(dest, overwrite: true);
```

Comment:
nit: handle directory creation with Nuke
```suggestion
            MoveFile(file, dest, FileExistsPolicy.Overwrite, createDirectories: true);
```